frontend/lib/src/components/elements/DeckGlJsonChart/withMapboxToken/withMapboxToken.tsx

```diff
@@ -63,25 +73,57 @@ const withMapboxToken =
         WrappedComponent.displayName || WrappedComponent.name
       })`
 
+      static contextType = LibContext
+
       public constructor(props: WrappedMapboxProps<P>) {
         super(props)
-
         this.state = {
           isFetching: true,
           mapboxToken: undefined,
           mapboxTokenError: undefined,
         }
+      }
+
+      /**
```

Comment:
Hard to read this as it's just added code... Here is original implementation:
```
/**
   * Expose a singleton MapboxToken:
   * - If the user specified a token in their streamlit config, return it.
   * - Else, fetch the remote "tokens.json" and return the "mapbox" entry.
   *
   * (The returned value is cached in memory, so the remote resource will
   * only be fetched once per session.)
   */
  public static async get(sessionInfo: SessionInfo): Promise<string> {
    const { commandLine, userMapboxToken } = sessionInfo.current

    if (
      !MapboxToken.token ||
      MapboxToken.commandLine !== commandLine.toLowerCase()
    ) {
      if (userMapboxToken !== "") {
        MapboxToken.token = userMapboxToken
      } else {
        MapboxToken.token = await this.fetchToken(TOKENS_URL, "mapbox")
      }

      MapboxToken.commandLine = commandLine.toLowerCase()
    }

    return MapboxToken.token
  }

  private static async fetchToken(
    url: string,
    tokenName: string
  ): Promise<string> {
    try {
      const response = await axios.get(url)
      const { [tokenName]: token } = response.data

      if (token == null || token === "") {
        throw new Error(`Missing token "${tokenName}"`)
      }

      return token
    } catch (e) {
      const error = ensureError(e)
      throw new MapboxTokenFetchingError(`${error.message} (${url})`)
    }
  }
```

The code above is combined into a single method and simplified.